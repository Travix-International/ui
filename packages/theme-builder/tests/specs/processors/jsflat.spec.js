const jsflatProcessor = require('../../../src/processors/jsflat.js');

describe('JS Flat object processor', () => {
  let doc;

  beforeEach(() => {
    doc = {
      do: {
        some: {
          string: 'aaa',
          num: '12px',
          color: '#fff',
        },
      },
    };
  });

  it('should compile doc to JS Flat object format', () => {
    const result = jsflatProcessor.compile(doc);
    expect(result).toMatchSnapshot();
  });

  it('should compile doc to JS Flat object format with prefix', () => {
    const result = jsflatProcessor.compile(doc, 'pre');
    expect(result).toMatchSnapshot();
  });
});
