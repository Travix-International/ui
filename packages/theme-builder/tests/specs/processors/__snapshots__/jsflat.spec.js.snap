// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JS Flat object processor should compile doc to JS Flat object format 1`] = `
Object {
  "do-some-color": "#fff",
  "do-some-num": "12px",
  "do-some-string": "aaa",
}
`;

exports[`JS Flat object processor should compile doc to JS Flat object format with prefix 1`] = `
Object {
  "pre-do-some-color": "#fff",
  "pre-do-some-num": "12px",
  "pre-do-some-string": "aaa",
}
`;
